node_modules/app/format_date.js

```diff
@@ -31,6 +30,10 @@ module.exports.format_due_date = function format_due (d) {
   return S.either(S.I, date => date.slice(0, ('/YYYY'.length)), format_date(d))
 }
 
+function date_to_string (d) {
+  return (new Date(d)).toISOString()
+}
+
 function check_date (date) {
-  return R.ifElse(Boolean, S.Right, S.Left)(date).chain(S.compose(S.maybeToEither(''), S.parseDate))
+  return R.ifElse(Boolean, S.Right, S.Left)(date).chain(S.compose(S.maybeToEither(''), R.compose(S.parseDate, date_to_string)))
```

Comment:
The issue with `date_to_string` is that `(new Date('invalid string')).toISOString()` will throw.

So let's think this through:

a) Date treats falsy arguments as no argument (returning a date for 01/01/1970), but we consider them bad
b) We get the string out of the date immediately in one case, and in the other we want the object. It's a toss up between what we return
c) We want to accept either `Date`s or strings

Simplest thing I think, is to use `S.encase` to deal with possible throwing, and just return the string

```js
return R.ifElse(Boolean, S.Right, S.Left)(date).chain(S.encase(R.constant(date), date_to_string))
```

then we can drop `.toISOString()` from `date_only`, but add `new Date` to the first `map` in `format_date`